src/callbacks_step/time_series_dg2d.jl

```diff
@@ -68,6 +70,52 @@ function get_elements_by_coordinates!(element_ids, coordinates, mesh::TreeMesh,
     return element_ids
 end
 
+# Elements on an `UnstructuredMesh2D` are possibly curved. Assume that each
+# element is convex, i.e., all interior angles are less than 180 degrees.
+# We use the barycenter of each element to determine if a given coordinate (x,y)
+# lies within said element. The shortest distance between the point and all the
+# barycenters "wins".
+function get_elements_by_coordinates!(element_ids, coordinates,
+                                      mesh::UnstructuredMesh2D,
+                                      dg, cache)
+    if length(element_ids) != size(coordinates, 2)
+        throw(DimensionMismatch("storage length for element ids does not match the number of coordinates"))
+    end
+
+    # Reset element ids - 0 indicates "not (yet) found"
+    element_ids .= 0
+
+    # Compute and save the barycentric coordinate on each element
+    bary_centers = zeros(eltype(mesh.corners), 2, mesh.n_elements)
+    distances = zeros(eltype(mesh.corners), mesh.n_elements)
+    calc_bary_centers!(bary_centers, dg, cache)
+
+    # Iterate over coordinates
+    for index in 1:length(element_ids)
+        # Compute the distance between the current point and all the bary centers.
+        for element in eachelement(dg, cache)
+            distances[element] = norm(coordinates[:, index] .- bary_centers[:, element])
+        end
+
+        # Locate the minimal distance as this gives the element containing the point
+        element_ids[index] = argmin(distances)
+    end
```

Comment:
Does something like this work as well? It should improve the performance a bit by avoiding allocations.

```suggestion
    # Compute and save the barycentric coordinate on each element
    bary_centers = zeros(eltype(mesh.corners), 2, mesh.n_elements)
    calc_bary_centers!(bary_centers, dg, cache)

    # Iterate over coordinates
    for index in 1:length(element_ids)
        point = SVector(coordinates[1, index], 
                        coordinates[2, index])
        
        # Search for the element with minimal distance between the point
        # we are interested in and the barycenter of the element.
        element_ids[index] = argmin(eachelement(dg, cache)) do element
            bary_center = SVector(bary_centers[1, element],
                                  bary_centers[2, element])
            # Compute the squared Euclidean distance between the point we are
            # interested in and the barycenter of the element.
            # We compute the squared norm to avoid computing computationally
            # more expensive square roots.
            sum(abs2, point - bary_center)
        end
    end
```

Reply:
I will give it a try. Patrick and I also discussed this strategy a bit and it is insufficient to identify the element correctly in certain common cases. So I need to change the strategy slightly anyway.